src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,52 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    public static final String PATTERN = "dd.MM.yyyy";
+    public static final String PARSING_ERROR = "Wrong input data";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+        LocalDate localDateFrom = getDateFrom(dateFrom);
+        LocalDate localDateTo = getDateFrom(dateTo);
+
+        for (String name : names) {
+            int salarySum = 0;
+            for (String dayData : data) {
+                String[] dayDataArray = dayData.split(" ");
+                LocalDate localDate = getDateFrom(dayDataArray[0]);
```

Comment:
It is a good practice to make all values constants in your code
```suggestion
                String[] dayDataArray = dayData.split(SEPARATOR);
                LocalDate localDate = getDateFrom(dayDataArray[DATE_INDEX]);
```